govcd/adminorg_ldap_test.go

```diff
@@ -0,0 +1,312 @@
+			AuthenticationMechanism: "SIMPLE",
+			ConnectorType:           "OPEN_LDAP",
+			Username:                "cn=admin,dc=planetexpress,dc=com",
+			Password:                "GoodNewsEveryone",
+			UserAttributes: &types.OrgLdapUserAttributes{
+				ObjectClass:               "inetOrgPerson",
+				ObjectIdentifier:          "uid",
+				Username:                  "uid",
+				Email:                     "mail",
+				FullName:                  "cn",
+				GivenName:                 "givenName",
+				Surname:                   "sn",
+				Telephone:                 "telephoneNumber",
+				GroupMembershipIdentifier: "dn",
+			},
+			GroupAttributes: &types.OrgLdapGroupAttributes{
+				ObjectClass:          "group",
+				ObjectIdentifier:     "objectGuid",
+				GroupName:            "cn",
+				Membership:           "member",
+				MembershipIdentifier: "dn",
+			},
+		},
+	}
+
+	err = org.LdapConfigure(ldapSettings)
+	check.Assert(err, IsNil)
+
+	fmt.Println(" Done")
+	AddToCleanupList("LDAP-configuration", "orgLdapSettings", org.AdminOrg.Name, check.TestName())
+}
+
+// createLdapServer spawns a vApp and photon OS VM. Using customization script it starts a testing
+// LDAP server in docker container which has a few users and groups defined.
+// In essence it creates two groups - "admin_staff" and "ship_crew" and a few users.
+// More information about users and groups in: https://github.com/rroemhild/docker-test-openldap
+func createLdapServer(vcd *TestVCD, check *C, directNetworkName string) (string, string, string) {
+	vAppName := "ldap"
+	// The customization script waits until IP address is set on the NIC because Guest tools run
+	// script and network configuration together. If the script runs too quick - there is a risk
+	// that network card is not yet configured and it will not be able to pull docker image from
+	// remote. Guest tools could also be interrupted if the script below failed before NICs are
+	// configured therefore it is run in background.
+	// It waits until "inet" (not "inet6") is set and then runs docker container
+	const ldapCustomizationScript = `
+		{
+			until ip a show eth0 | grep "inet "
+			do
+				sleep 1
+			done
+			systemctl enable docker
+			systemctl start docker
+			docker run --name ldap-server --restart=always --privileged -d -p 389:389 rroemhild/test-openldap
+		} &
+	`
+	// Get Org, Vdc
+	org, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	vdc, err := org.GetVDCByName(vcd.config.VCD.Vdc, false)
+	check.Assert(err, IsNil)
+	check.Assert(vdc, NotNil)
+
+	// Find catalog and catalog item
+	catalog, err := org.GetCatalogByName(vcd.config.VCD.Catalog.Name, false)
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+	catalogItem, err := catalog.GetCatalogItemByName(vcd.config.VCD.Catalog.CatalogItem, false)
+	check.Assert(err, IsNil)
+	// Skip the test if catalog item is not Photon OS
+	if !isItemPhotonOs(*catalogItem) {
```

Comment:
This check should happen in front of all testing, as major thing to have to run the test.

Reply:
But that is exactly the same as in `createAngGetResourcesForVmCreation`. I think readability wise I will pollute the environment if pull it out.

Reply:
Let's @dataclouder  decide. If we need less running test with caveats.

Reply:
If we need to run this check, we should either bail out as early as possible, or reorganize the auxiliary functions to return an error, allowing the callers to clean up in case of problems.
As it is now, an error (or a skip) may leave the system in an unclean state

Reply:
let it be. moved out